math/iterative_factorial.cpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file
+ * @brief Iterative implementation of [Factorial](https://en.wikipedia.org/wiki/Factorial)
+ *
+ * @author [Renjian-buchai](https://github.com/Renjian-buchai)
```

Comment:
bit of a nit pick but would you mind moving the author tag to the bottom as is tradition